apps/src/common.rs

```diff
@@ -1640,3 +1649,1002 @@ impl HttpConn for Http3Conn {
+                        ),
+                        quiche::h3::Header::new(b"server", b"quiche"),
+                    ];
+
+                    return Ok((
+                        headers,
+                        b"Invalid scheme".to_vec(),
+                        Default::default(),
+                    ));
+                }
+
+                scheme
+            },
+
+            None =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":scheme cannot be missing".to_string(),
+                )),
+        };
+
+        let decided_host = match (authority, host) {
+            (None, Some("")) =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    "host value cannot be empty".to_string(),
+                )),
+
+            (Some(""), None) =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":authority value cannot be empty".to_string(),
+                )),
+
+            (Some(""), Some("")) =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":authority and host value cannot be empty".to_string(),
+                )),
+
+            (None, None) =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":authority and host missing".to_string(),
+                )),
+
+            // Any other combo, prefer :authority
+            (..) => authority.unwrap(),
+        };
+
+        let decided_path = match path {
+            Some("") =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":path value cannot be empty".to_string(),
+                )),
+
+            None =>
+                return Err((
+                    H3_MESSAGE_ERROR,
+                    ":path cannot be missing".to_string(),
+                )),
+
+            Some(path) => path,
+        };
+
+        let url = format!("{decided_scheme}://{decided_host}{decided_path}");
+        let url = url::Url::parse(&url).unwrap();
+
+        let pathbuf = path::PathBuf::from(url.path());
+        let pathbuf = autoindex(pathbuf, index);
+
+        // Priority query string takes precedence over the header.
+        // So replace the header with one built here.
+        let query_priority = priority_field_value_from_query_string(&url);
+
+        if let Some(p) = query_priority {
+            priority = p.as_bytes().to_vec();
+        }
+
+        let (status, body) = match decided_method {
+            "GET" => {
+                for c in pathbuf.components() {
+                    if let path::Component::Normal(v) = c {
+                        file_path.push(v)
+                    }
+                }
+
+                match std::fs::metadata(file_path.as_path()) {
+                    Ok(metadata) => {
+                        (200, std::iter::repeat(1).take(metadata.len().try_into().unwrap()).collect())
```

Comment:
For fair comparison, whatever you end up sending should also be used when doing non-multicore benchmark